### ISSUE TITLE
[$250] Hybrid app - "Welcome to Expensify" modal when signing into the existing account on a new device.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49611]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.39-0
**Reproducible in staging?:** Y
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1726864963819969

## Action Performed:
1. Go to staging.new.expensify.com and create an account (Any other platform)
2. Signin with the same credentials on Hybrid app
3. Tap Try New Expensify from menu
## Expected Result:
No welcome modal as user already signed in another platform
## Actual Result:
"Welcome to Expensify" modal when signing in
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<img src="https://github.com/user-attachments/assets/2c0a99d2-1e3b-4a58-b8f5-df2b8f5cabc8" width="40%" height="40%">

<img src="https://github.com/user-attachments/assets/184cd264-5ae7-4798-ab98-c7c57ab879d2" width="40%" height="40%">

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021838490385865012073</li>
        <li>Upwork Job ID: 1838490385865012073</li>
        <li>Last Price Increase: 2025-01-28</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alitoshmatov</details>

**Comment by klajdipaja** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The user has already logged into the New Expensify on a web browser but when he logs into the hybrid app he is shown the welcome to new Expensify modal even though he has already been using it in the web.

### What is the root cause of that problem?
We have not taken into account that the user might login first through the web and then in the app. The functions that check if the user should see the welcome modal use the onyx key `nvp_tryNewDot.classicRedirect.completedHybridAppOnboarding` which is only set when the user logs on the native app on the first time.

### What changes do you think we should make in order to solve the problem?
We can use the `nvp_onboarding.hasCompletedGuidedSetupFlow` flag from onyx to detect that the user has already used the NewDot. This flag when an account is used in the web for the first time is set to to true but on the native app is set to true only once the guided welcome flow has completed. If we add this into the `isFirstTimeHybridAppUser` function in https://github.com/Expensify/App/blob/main/src/libs/actions/Welcome/index.ts we can make sure that the welcome message does not show. 
The first check in that function should be the following:

```
   const hasCompletedGuidedSetupFlow = !(Array.isArray(onboarding) || onboarding?.hasCompletedGuidedSetupFlow === undefined) && onboarding?.hasCompletedGuidedSetupFlow
        if (hasCompletedGuidedSetupFlow) {
            return;
        }


```

We can also put this block before `isFirstTimeHybridAppUser` is called:

```
function handleHybridAppOnboarding() {
    if (!NativeModules.HybridAppModule) {
        return;
    }
       const hasCompletedGuidedSetupFlow = !(Array.isArray(onboarding) || onboarding?.hasCompletedGuidedSetupFlow === undefined) && onboarding?.hasCompletedGuidedSetupFlow
        if (hasCompletedGuidedSetupFlow) {
            return;
        }
    
```

### What alternative solutions did you explore? (Optional)

**Comment by bfitzexpensify** [comment]
Proposal ready for review in https://github.com/Expensify/App/issues/49611#issuecomment-2370707723 @alitoshmatov

**Comment by alitoshmatov** [comment]
I couldn't reproduce it in our existing app. As far as I understand this is only happening in hybrid app(which I don't have much info about). But based on [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1727972599140359?thread_ts=1727971011.835919&cid=C01GTK53T8Q) C+ should get access soon. 

@bfitzexpensify I am not sure what is the next step. If it is not an urgent issue I think we can hold till C+ get access and we can continue then

**Comment by bfitzexpensify** [comment]
>If it is not an urgent issue I think we can hold till C+ get access and we can continue then

Agreed. Setting to weekly and starring myself to keep checking in on this. We can revisit once C+ have access

**Comment by trjExpensify** [comment]
@alitoshmatov you should be able to download the HybridApp from the Apple App store if you go to the "Expensify" listing. 

I'm not sure if anyone will run into this bug now in practice, because if they sign-up on web either on new.expensify.com or expensify.com and meet the conditions to get redirected into NewDot, the tryNewDot.classicRedirect.dismissed NVP will be set to `false`. Meaning, the next visit on mobile will launch NewDot not OldDot and require clicking the "Try New Expensify" button. 
 
To illustrate that:

- Sign-up on expensify.com with a new account 
- Choose the "1-9" SignUpQualifier, so you get redirected to NewDot and the tryNewDot classicRedirect.dismissed NVP set to false
- Complete the onboarding modal steps 
- Go to the Apple App Store and download the latest version available of the "Expensify" app (not the "New Expensify" app).
- Log-in with those credentials 
- The NewDot experience will load

CC: @Julesssss @AndrewGable

**Comment by alitoshmatov** [comment]
> I'm not sure if anyone will run into this bug now in practice

I agree with [this](https://github.com/Expensify/App/issues/49611#issuecomment-2416263702). I don't think this normal user will come across this issue

**Comment by trjExpensify** [comment]
@puneetlath I see you reported this one. Any thoughts on what I said [here](https://github.com/Expensify/App/issues/49611#issuecomment-2416263702) before we potentially close this out?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)
